#!/usr/bin/env python
# DWA Planner configuration

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, double_t, int_t, bool_t
gen = ParameterGenerator()

gen.add("min_linear_velocity", double_t, 0, "The minimum desired linear velocity for the robot. [m/s]", 0.3, 0, 1.5)
gen.add("nominal_linear_velocity", double_t, 0, "The nominal desired linear velocity for the robot. [m/s]", 1.0, 0, 1.5)

gen.add("min_range_front", double_t, 0, "The minimum range that obstacles effect the robot. [m]", 0.1, 0, 2.0)
gen.add("nominal_range_min_front", double_t, 0, "The minimum range of obstacles to use the nominal velocity. [m]", 0.3, 0, 3.0)
gen.add("nominal_range_max_front", double_t, 0, "The maximum range of obstacles to use the nominal velocity. [m]", 0.7, 0, 3.0)
gen.add("max_range_front", double_t, 0, "The maximum range that obstacles effect the robot's velocity. [m]", 1.0, 0, 3.0)

gen.add("min_range_rear", double_t, 0, "The minimum range that obstacles effect the robot. [m]", 0.1, 0, 2.0)
gen.add("nominal_range_min_rear", double_t, 0, "The minimum range of obstacles to use the nominal velocity. [m]", 0.3, 0, 3.0)
gen.add("nominal_range_max_rear", double_t, 0, "The maximum range of obstacles to use the nominal velocity. [m]", 0.7, 0, 3.0)
gen.add("max_range_rear", double_t, 0, "The maximum range that obstacles effect the robot's velocity. [m]", 1.0, 0, 3.0)

gen.add("min_angular_velocity", double_t, 0, "The minimum desired angular velocity for the robot. [rad/s]", 0.6, 0, 2.0)
gen.add("min_angular_velocity_effect_distance", double_t, 0, "The minimum range beyond the circumscribed radius to add for slowing down. [m]", 0.2, 0, 2.0)
gen.add("max_angular_velocity_effect_distance", double_t, 0, "The maximum range beyond the circumscribed radius to add for slowing down. [m]", 0.5, 0, 2.0)
gen.add("wally_speed_profile", bool_t, 0, "Enable the wally speed profile curve", False)
gen.add("extended_obstacle_curve", double_t, 0, "Curve shape for new obstacle curve: pow(distance, 1/(param))", 1.8, 0.001, 3.0)

exit(gen.generate("base_local_planner", "base_local_planner", "StaticSpeedLimiter"))
